### PR TITLE
Wip0.0.1

### PR DESCRIPTION
**Pull request by jvlker**
Adding feedback of last review to the service:
- remove feedgroups
- remove POST /animals
- remove history_ids
- add some filter-parameters

### CODE REVIEW

File: animal_data_oas.yml [hunk 1]
```diff
@@ -0,0 +1,574 @@
+openapi: '3.0.3'
```
Comment (agricircle) [inline animal_data_oas.yml]: Let's use the same convention of naming the main OpenAPI file 'oas' as in all other templates. The 'oas' file is already present and populated with the latest specs you had sent me

File: README.md [hunk 2]
```diff
@@ -1 +1,17 @@
-# animal_data
\ No newline at end of file
+# animal_data
+
+[OpenAPI Specifications](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/fodjan/atlas_animal_data/wip0.0.1/animal_data_oas.yml)
+
+
+# ATLAS Service Template Specification
+
+| | |
```
Comment (agricircle) [inline README.md]: It's ok to copy the abstract from the doc.html in the readme, but including the "header" with version number will force us to remember to update this file as well, in addition to doc.html and oas, each time a new version is done.

**Review by agricircle — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jvlker) [reply README.md]: ok, my plan was to put the doc in the readme. But with your version we don't need that.

File: oas [hunk 3]
```diff
@@ -2,11 +2,12 @@ openapi: '3.0.3'
 info:
   title: ATLAS Service Template - Animal data
   description: Setup and synchronize the animals of a farm and their organization in groups and pens.
-  version: 0.0.1
+  version: 0.0.5
```
Comment (agricircle) [inline oas]: The OAS version number must match the template version number. The ATLAS Registry enforces strict semantic versioning and will reject any new version that is not either:
old_major+1.0.0
old_major.old_minor+1.0
old_major.old_minor.old_patch+1

In this case, the release will be either 0.0.1 to indicate the first experimental release, or 0.1.0

File: oas [hunk 4]
```diff
@@ -395,39 +352,7 @@ components:
...
-        - name
-      example:
-        urn: 'urn:animal_service:FDBE6F55-6179-4B88-9177-4AFBED33ADAC'
-        date_created: '2022-03-14T00:00:00'
-        date_last_update: '2022-05-04T00:00:42'
-        archived: false
-        name: 'Highyield 1'
-        description: '1st Pen new stable left side'
+      description: 'Uniform resource name of a field with the syntax `urn:<service_id>:<internal_id>`. <service_id> must be the urn of a valid and active ATLAS service.'
     Barn:
```
Comment (agricircle) [inline oas:359]: We are missing an endpoint that will allow us to find the pen an animal was in at a particular date.
Also, is it possible that an animal is not associated to a pen for a period of time?

File: oas [hunk 5]
```diff
@@ -395,39 +352,7 @@ components:
...
       type: object
       properties:
```
Comment (agricircle) [inline oas:364]: optional geometry is fine and useful but there should probably be a GPS location for the barn that is mandatory to track physical location of animals in general

**Review by agricircle — CHANGES_REQUESTED** [review]

File: oas (hunk 4 above)
Comment (jvlker) [reply oas:359]: 1. You can query the animal at a particular date and use the pen_url out of that response: `GET /animals/{urn}?date=2020-01-01`
Additionally I think it is useful to get all animals in a given pen (at a certain date). So I added a new filter: `GET /animals?date=2020-01-01&pen_urn=...`
2. Indeed when animals are at the pasture there is no pen-assignment for that time. To support that usecase I would suggest to remove the required for animal/pen_urn. They are still assigned to a farm an can be accessed the normal way over `GET /animals`

File: oas (hunk 5 above)
Comment (jvlker) [reply oas:364]: - Isn't GeoJSON exactly specified for that? I have no experience how such locations are usually handled but the GeoJSON "Point" tuple seams exactly what you're looking for? Or do you prefer extra longitute/latitute fields?
- Are you sure you want this information required? Not all usecases have or need this information. For example we don't need this information to feed the animals.

File: README.md [hunk 6]
```diff
@@ -1 +1,17 @@
...
+| --- | --- |
+| Name | animal_data |
+| Version | 0.2.6 |
+| Status | DRAFT |
```
Comment (agricircle) [inline README.md]: The 'Status' has been removed from all specification. The fact that it is a DRAFT is made clear by the branch it is in

File: oas (hunk 4 above)
Comment (agricircle) [reply oas:359]: Makes sense. I suggest that (1) is clearly stated in the endpoint description. Regarding 2., can you make the change in the OAS (not required, and explain the meaning of a missing or null pen_urn)

File: oas (hunk 5 above)
Comment (agricircle) [reply oas:364]: What I meant is that while a more detailed geometry (e.g. polygon) may be only optional, a point location should be mandatory (in lieu of a street address)

File: oas [hunk 7]
```diff
@@ -487,8 +414,6 @@ components:
         max_number_animals:
```
Comment (agricircle) [inline oas]: by the way, if we specify the max number of animals, I guess we should also specify what kind of animals are associated to a pen at a given point of time

**Review by agricircle — CHANGES_REQUESTED** [review]

File: oas (hunk 5 above)
Comment (jvlker) [reply oas:364]: I included geo_location and geo_layout for barn and pen as optional fields.

File: README.md (hunk 6 above)
Comment (jvlker) [reply README.md]: doc is updated

File: oas (hunk 7 above)
Comment (jvlker) [reply oas]: removed max_number_animals.

File: oas [hunk 8]
```diff
@@ -445,8 +378,10 @@ components:
           type: string
         description:
           type: string
-        geometry:
-          $ref: 'https://geojson.org/schema/GeoJSON.json'
+        geo_location:
+          $ref: 'https://geojson.org/schema/Point.json'
+        geo_layout:
+          $ref: 'http://geojson.org/schema/Polygon.json'
```
Comment (agricircle) [inline oas]: Should be https, otherwise fetching fails in swagger editor 
image

File: oas [hunk 9]
```diff
@@ -507,14 +441,19 @@ components:
         name: 'Highyield 1'
         description: '1st Pen new stable left side'
         max_number_animals: 60
```
Comment (agricircle) [inline oas]: Should be removed

File: oas [hunk 10]
```diff
@@ -147,39 +141,7 @@ paths:
...
-        200:
-          description: 'successful operation'
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Feedgroup'
+
 
   /subscriptions:
     post:
```
Comment (agricircle) [inline oas:145]: In a fundamental service such as animal_data, I believe that support for subscriptions really should be mandatory toi avoid constant polling from all subscribers, like it is in field_data for fields/groups.

**Review by agricircle — CHANGES_REQUESTED** [review]

File: oas (hunk 10 above)
Comment (jvlker) [reply oas:145]: ok, makes sense. I removed the optional keywords in descriptions and doc.

**Review by agricircle — APPROVED** [review]

**Pull request merged**